flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -169,9 +169,8 @@ void retriggerSubpartitionRequest(int subpartitionIndex) throws IOException {
 
 	@Override
 	Optional<BufferAndAvailability> getNextBuffer() throws IOException {
-		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a queue.");
-
 		checkError();
+		checkState(partitionRequestClient != null, "Queried for a buffer before requesting a queue.");
```

Comment:
Have you thought about other places that we are doing
```
checkState(partitionRequestClient != null, ...);
```
? Frankly at the first glance it looks like everywhere we should be doing `checkError` before. If that's correct, maybe we should introduce some helper method 
```
checkPartitionRequestQueueInitialized() {
  checkError();
  checkState(partitionRequestClient != null, ...);
}
```
?

Reply:
@pnowojski Yes. I've checked other places including :

* notifyCreditAvailable: called from netty thread after the connection is built.
* resumeConsumption: called when receiving barriers.
* retriggerPartitionRequest: called after the connection is built. 
* sendTaskEvent: the exception will be thrown directly to StreamTask / Environment

If I understand correctly, none of them will cause this issue. But it's also reasonable to add this additional check and bind them together because we may introduce other changes that cause this.

Reply:
I think you are right, that it should be fine for now, but as you wrote:
> But it's also reasonable to add this additional check and bind them together because we may introduce other changes that cause this in the future.

it's probably better to future proof those checks and bind them together.